mailpoet/lib/Entities/StatisticsNewsletterEntity.php

```diff
@@ -36,19 +36,21 @@ class StatisticsNewsletterEntity {
   private $subscriber;
 
   /**
-   * @ORM\Column(type="datetimetz", nullable=true)
-   * @var \DateTimeInterface|null
+   * @ORM\Column(type="datetimetz", nullable=false)
+   * @var \DateTimeInterface
    */
   private $sentAt;
 
   public function __construct(
     NewsletterEntity $newsletter,
     SendingQueueEntity $queue,
-    SubscriberEntity $subscriber
+    SubscriberEntity $subscriber,
+    \DateTimeInterface $sentAt = null
   ) {
     $this->newsletter = $newsletter;
     $this->queue = $queue;
     $this->subscriber = $subscriber;
+    $this->sentAt = $sentAt ?: new \DateTimeImmutable();
```

Comment:
Sometimes we use `Carbon::createFromTimestamp($this->wp->currentTime('timestamp'))`. But I'm not sure if that's needed at all. Is there a difference?

Reply:
I pass `Carbon::createFromTimestamp($this->wp->currentTime('timestamp'))` date into the constructor but as a default value (used now only in tests) I used the DateTimeImmutable because I didn't want to add a dependency on a WP function into the entity constructor.